### PR TITLE
refine job management and scheduler service

### PR DESCRIPTION
**Pull request by hongyh13**

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomSchedulerService.cs [hunk 1]
```diff
@@ -411,30 +405,32 @@ private async Task CheckAndUpdateOrchestratorJobStatusAsync(CurrentTriggerEntity
         /// <returns>return true if the job has been scheduled to end, otherwise return false.</returns>
         private async Task<bool> CreateNextTriggerAsync(CurrentTriggerEntity currentTriggerEntity, CancellationToken cancellationToken)
         {
-            DateTimeOffset? nextTriggerEndTime = GetNextTriggerEndTime(currentTriggerEntity.TriggerEndTime);
+            DateTimeOffset? nextTriggerTime = GetNextTriggerTime(currentTriggerEntity.TriggerEndTime);
 
-            if (nextTriggerEndTime != null)
+            if (nextTriggerTime != null )
```
Comment (BoyaWu10) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomSchedulerService.cs:410]: Nit: extra space here.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomToDataLakeOrchestratorJob.cs [hunk 2]
```diff
@@ -54,10 +50,8 @@ public class DicomToDataLakeOrchestratorJob : IJob
             _jobInfo = EnsureArg.IsNotNull(jobInfo, nameof(jobInfo));
             _inputData = EnsureArg.IsNotNull(inputData, nameof(inputData));
             _result = EnsureArg.IsNotNull(result, nameof(result));
-            _dataClient = EnsureArg.IsNotNull(dataClient, nameof(dataClient));
             _dataWriter = EnsureArg.IsNotNull(dataWriter, nameof(dataWriter));
             _queueClient = EnsureArg.IsNotNull(queueClient, nameof(queueClient));
-            _metadataStore = EnsureArg.IsNotNull(metadataStore, nameof(metadataStore));
```
Comment (BoyaWu10) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomToDataLakeOrchestratorJob.cs:60]: Nit: Can you help to update the logs in orchestration job and processing job, like from `FhirToDataLake` to `DicomToDataLake`? So we can avoid modifying same file.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomSchedulerService.cs (hunk 1 above)
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomSchedulerService.cs:410]: Fixed.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomSchedulerService.cs [hunk 3]
```diff
@@ -447,12 +443,21 @@ private async Task<bool> CreateNextTriggerAsync(CurrentTriggerEntity currentTrig
...
+                return null;
+            }
+
             var initialTriggerEntity = new CurrentTriggerEntity
             {
                 PartitionKey = TableKeyProvider.TriggerPartitionKey(_queueType),
                 RowKey = TableKeyProvider.TriggerPartitionKey(_queueType),
-                TriggerStartTime = GetNextTriggerStartTime(null),
-                TriggerEndTime = (DateTimeOffset)GetNextTriggerEndTime(null),
+                TriggerEndTime = DateTimeOffset.Now,
```
[the remark this answers is not in the document]
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomSchedulerService.cs:460]: Sure, created a new queue type `DicomToDataLake`.

[the remark this answers is not in the document]
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomSchedulerService.cs:460]: `latestSequence` is used for `EndOffset`

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomToDataLakeOrchestratorJob.cs (hunk 2 above)
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/DicomToDataLakeOrchestratorJob.cs:60]: Updated.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/DicomToDataLakeProcessingJobResult.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Jobs.Models
+{
+    public class DicomToDataLakeProcessingJobResult
```
Comment (BoyaWu10) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/DicomToDataLakeProcessingJobResult.cs:11]: Is this one exactly the same with the one for FHIR? If so, can we make a common class for both data sources?

Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/DicomToDataLakeProcessingJobResult.cs:11]: No, it is different. we don't need patient version information in `DicomToDataLakeProcessingJobResult`

Comment (BoyaWu10) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/DicomToDataLakeProcessingJobResult.cs:11]: Got it. Thanks!

**Review by BoyaWu10 — APPROVED** [review]

**Pull request merged**